fe/fe-core/src/main/java/org/apache/doris/datasource/property/fileformat/CsvFileFormatProperties.java

```diff
@@ -0,0 +1,195 @@
+        this.textSerdeType = textSerdeType;
+    }
+
+    public CsvFileFormatProperties(TFileFormatType fileFormatType, String headerType) {
+        super(fileFormatType);
+        this.headerType = headerType;
+    }
+
+
+    @Override
+    public void analyzeFileFormatProperties(Map<String, String> formatProperties, boolean isRemoveOriginProperty)
+            throws AnalysisException {
+        try {
+            // check properties specified by user -- formatProperties
+            columnSeparator = getOrDefaultAndRemove(formatProperties, CsvProperties.PROP_COLUMN_SEPARATOR,
+                    defaultColumnSeparator, isRemoveOriginProperty);
+            if (Strings.isNullOrEmpty(columnSeparator)) {
+                throw new AnalysisException("column_separator can not be empty.");
+            }
+            columnSeparator = Separator.convertSeparator(columnSeparator);
+
+            lineDelimiter = getOrDefaultAndRemove(formatProperties, CsvProperties.PROP_LINE_DELIMITER,
+                    CsvProperties.DEFAULT_LINE_DELIMITER, isRemoveOriginProperty);
+            if (Strings.isNullOrEmpty(lineDelimiter)) {
+                throw new AnalysisException("line_delimiter can not be empty.");
+            }
+            lineDelimiter = Separator.convertSeparator(lineDelimiter);
+
+            String enclosedString = getOrDefaultAndRemove(formatProperties, CsvProperties.PROP_ENCLOSE,
+                    "", isRemoveOriginProperty);
+            if (!Strings.isNullOrEmpty(enclosedString)) {
+                if (enclosedString.length() > 1) {
+                    throw new AnalysisException("enclose should not be longer than one byte.");
+                }
+                enclose = (byte) enclosedString.charAt(0);
+                if (enclose == 0) {
+                    throw new AnalysisException("enclose should not be byte [0].");
+                }
+            }
+
+            trimDoubleQuotes = Boolean.valueOf(getOrDefaultAndRemove(formatProperties,
+                    CsvProperties.PROP_TRIM_DOUBLE_QUOTES, "", isRemoveOriginProperty))
+                    .booleanValue();
+            skipLines = Integer.valueOf(getOrDefaultAndRemove(formatProperties,
+                    CsvProperties.PROP_SKIP_LINES, "0", isRemoveOriginProperty)).intValue();
+            if (skipLines < 0) {
+                throw new AnalysisException("skipLines should not be less than 0.");
+            }
+
+            String compressTypeStr = getOrDefaultAndRemove(formatProperties,
+                    CsvProperties.PROP_COMPRESS_TYPE, "UNKNOWN", isRemoveOriginProperty);
+            try {
+                compressionType = Util.getFileCompressType(compressTypeStr);
+            } catch (IllegalArgumentException e) {
+                throw new AnalysisException("Compress type : " +  compressTypeStr + " is not supported.");
+            }
+
+            FileFormatUtils.parseCsvSchema(csvSchema, getOrDefaultAndRemove(formatProperties,
```

Comment:
I think `csv_schema` is not part of `CsvFileFormatProperties`. It is part of `tvf` properties.